lib/nose/plans/index_lookup.rb

```diff
@@ -121,6 +125,36 @@ def self.check_parent_index(parent, index, state)
       end
       private_class_method :check_parent_index
 
+      # If one column family used as materialized view and second step of query plan,
+      # costs of each column family differs because of the cost of second step changes according to the cardinality of the first step.
+      # Validation process fails because of two different cost of the same column family.
+      # This happens when one query has equality condition of id field and non-id field.
+      def self.is_useless_parent?(state,index,parent_index)
+
+        #SELECT * FROM entity WHERE A = ? AND B = ?
+        # parent: [A][B]->[C,D]
+        # index:  [B][A]->[C,D,E]
+        return true if index.extra >= parent_index.extra and \
+                          state.query.eq_fields >= (parent_index.hash_fields + parent_index.order_fields.to_set) and \
+                          parent_index.hash_fields == index.order_fields.to_set and \
+                          parent_index.order_fields.to_set == index.hash_fields
+
+        #SELECT * FROM entity WHERE A = ? AND B = ?
+        # parent: [A][B]->[C,D]
+        # index:  [A,B][F]->[C,D,E]
+        return true if index.hash_fields >= state.query.eq_fields and \
+                       index.all_fields >= parent_index.all_fields
+
+        #SELECT E FROM entity WHERE A = ? AND B = ?
+        # parent: [A,B][]->[C,D]
+        # index:  [A][B]->[E]
```

Comment:
If I understand correctly, `C`, `D`, and `E`, and `F` don't actually matter here. I would replace them with `[...]` to make this clear (same with any similar cases above). If they actually do matter, it seems like there should be a condition on `extra`.